src/machine/machined-dbus.c

```diff
@@ -599,7 +599,8 @@ static int method_get_image_os_release(sd_bus_message *message, void *userdata,
         return redirect_method_to_image(message, userdata, error, bus_image_method_get_os_release);
 }
 
-static int clean_pool_done(Operation *operation, int ret, sd_bus_error *error) {
+static int clean_pool_done(Operation *operation, int ret, void *userdata) {
```

Comment:
why change the 3rd paramater type here?

i think it's fine to leave the parameter as is. sd_bus_error stuff is supposed to be *optional* anyway, when pass to functions. The idea is to fill it in, if it is passed non-NULL, but not bother if it is passed as NULL.

Hence, following that logic, I think these functions should systematically take an sd_bus_error and just not bother if it#s NULL, and then use that fact for the varlink code paths

Reply:
but as mentioned i think it's fine to just leave theoriginal type in place. just pass NULL to the sd_bus_error object if you cal this function outside of dbus context